src/components/ToolBar/DataMenu/UploadNetworkMenuItem.tsx

```diff
@@ -0,0 +1,147 @@
+
+  const setTables = useTableStore((state) => state.setTables)
+
+  const addNetworkToWorkspace = useWorkspaceStore(
+    (state) => state.addNetworkIds,
+  )
+  const createDataFromLocalCx2 = async (
+    LocalNetworkId: string,
+    cxData: Cx2,
+  ): Promise<FullNetworkData> => {
+    const network: Network = NetworkFn.createNetworkFromCx(LocalNetworkId, cxData)
+    await putNetworkToDb(network)
+
+    const [nodeTable, edgeTable]: [Table, Table] = TableFn.createTablesFromCx(
+      LocalNetworkId,
+      cxData,
+    )
+    await putTablesToDb(LocalNetworkId, nodeTable, edgeTable)
+
+    const visualStyle: VisualStyle = VisualStyleFn.createVisualStyleFromCx(cxData)
+    await putVisualStyleToDb(LocalNetworkId, visualStyle)
+  
+    const networkView: NetworkView = ViewModelFn.createViewModelFromCX(
+      LocalNetworkId,
+      cxData,
+    )
+    await putNetworkViewToDb(LocalNetworkId, networkView)
+  
+    return { network, nodeTable, edgeTable, visualStyle, networkView }
+  }
+  
+  
+    const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+      const file = event.target.files?.[0];
+      if (file == null) return;
+      const reader = new FileReader();
+      reader.onload = async (event) => {
+        try {
+          const json = JSON.parse(event.target?.result as string);
+          const localName = json[3].networkAttributes[0].name
+          const localDescription = json[3].networkAttributes[0].description
+          const localUuid = uuidv4();
+          const localNodeCount = json[1].metaData[3].elementCount
+          const localEdgeCount = json[1].metaData[2].elementCount
+          await putNetworkSummaryToDb({
+            "ownerUUID": localUuid, "name": localName,
+            isReadOnly: false,
+            subnetworkIds: [],
+            isValid: false,
+            warnings: [],
+            isShowcase: false,
+            isCertified: false,
+            indexLevel: '',
+            hasLayout: false,
+            hasSample: false,
+            cxFileSize: 0,
+            cx2FileSize: 0,
+            properties: [],
+            owner: '',
+            version: '',
+            completed: false,
+            visibility: 'PUBLIC',
+            nodeCount: localNodeCount,
+            edgeCount: localEdgeCount,
+            description: localDescription,
+            creationTime: new Date(Date.now()),
+            externalId: localUuid,
+            isDeleted: false,
+            modificationTime: new Date(Date.now())
+          })
+          const res = await createDataFromLocalCx2(localUuid, json)
+          const { network, nodeTable, edgeTable, visualStyle, networkView } = res
+          addNetworkToWorkspace(localUuid)
+          addNewNetwork(network)
+          setVisualStyle(localUuid, visualStyle)
+          setTables(localUuid, nodeTable, edgeTable)
+          setViewModel(localUuid, networkView)
+          setTimeout(() => {
+            setCurrentNetworkId(localUuid)
```

Comment:
Don't need to use setTimeout when setting current network id anymore.  The bug related to this has been fixed.